Fix TypeError with toplist modal

Animus-Surge opened the pull request:
Mentioned in #929 

Fixing minor error where uncaught TypeErrors would show in the developer console in regards to the toplist class trying to update the contents of the modal when the modal is not present.

KjartanOli commented:
Would it not be better to use clearInterval to stop the function from executing at all? It probably doesn't make much of a difference, but I would imagine it was preferable not to call a function that does nothing every second.

GitSquared commented:
Agreed, we should use a deconstructor of some sort to fix this.

Animus-Surge commented:
> Would it not be better to use clearInterval to stop the function from executing at all?

Alright, changed how checking is handled. Only problem I have is the error now shows up, though it's only sent once or twice and then stops and that might be the way I implemented it.

KjartanOli commented:
src/classes/modal.class.js
```js
        this.close = () => {
            let modalElement = document.getElementById("modal_"+this.id);
            modalElement.setAttribute("class", "modal_popup "+this.type+" blink");
            window.audioManager.denied.play();
            setTimeout(() => {
                modalElement.remove();
                delete window.modals[this.id];
            }, 100);

            if (typeof this.onclose === "function") {
                this.onclose();
            }
        };
```

Looks like the modal itself gets deleted before clearInterval is called. It might be possible to fix this by just moving the if statement, but I don't know the rest of the codebase, so I don't know if some other modal relies on things being done in this order.

Animus-Surge commented:
Hmm, I tried this by moving the if statement to the top of the function and it ended up doing the exact same thing. I tried something else (which ends up using another variable), checking if the modal had been removed already, and if it has it'll clear the interval. I added an else statement to ensure no errors show up as well.

praniltakawane reviewed (commented):
Hi

scientistobse commented:
Hello

On Wed, Dec 2, 2020, 4:03 PM Pranil Takawane <notifications@github.com>
wrote:

> *@praniltakawane* commented on this pull request.
>
> Hi
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GitSquared/edex-ui/pull/930#pullrequestreview-542700215>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQYDSUHQNWWGTYTCJFJF2KLSSYJY5ANCNFSM4UJIZVGA>
> .
>

GitSquared commented:
Damned spams.

GitSquared reviewed (commented):
Looks like we're good to go!

razaamir commented:
Hello, how are you?

Raza

On Wed, Dec 2, 2020 at 3:35 PM scientistobse <notifications@github.com>
wrote:

> Hello
>
> On Wed, Dec 2, 2020, 4:03 PM Pranil Takawane <notifications@github.com>
> wrote:
>
> > *@praniltakawane* commented on this pull request.
> >
> > Hi
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/GitSquared/edex-ui/pull/930#pullrequestreview-542700215
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AQYDSUHQNWWGTYTCJFJF2KLSSYJY5ANCNFSM4UJIZVGA
> >
> > .
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/GitSquared/edex-ui/pull/930#issuecomment-737143508>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHGK5TCTXPTQ2XSO62ZEZKLSSYKAPANCNFSM4UJIZVGA>
> .
>
-- 
Raza Amir

GitSquared commented:
F*** off.
